sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -503,6 +503,12 @@ public void close() {
         return createMessageBatch().flatMap(messageBatch -> {
             messages.forEach(message -> messageBatch.tryAddMessage(message));
             return sendInternal(messageBatch, transaction);
+        }).onErrorMap(throwable -> {
```

Comment:
This mapping logic is in many places. It would be better to have a static method that this calls into which maps a throwable into the corresponding ServiceBusErrorSource.

Reply:
Moved this into same class for reuse.